### ISSUE TITLE
Crashes when no monitors are left

### ISSUE THREAD

**Issue by haarp** [rust-windowing/winit#1314]
(opening this in favor of rust-windowing/glutin#1247)

Gentoo Linux X11, Xfwm4

Hello,

when disconnecting all monitors from my laptop and then closing the lid, there is a high probability that all glutin-based windows crash. I am experiencing this with alacritty 0.4.0 and glutin's example window (`cargo run --example window`) with sources as of rust-windowing/glutin@a085d7c.

Curiously, this doesn't seem to happen when I unplug the last monitor after closing the lid. Relevant log:

```thread 'main' panicked at 'Failed to update normal hints: XError { description: "BadValue (integer parameter out of range for operation)", error_code: 2, request_code: 12, minor_code: 0 }', src/libcore/result.rs:999:5```

Also see jwilm/alacritty#3076

Thanks!

**Comment by murarth** [comment]
Can you test this using the latest winit master? We recently merged a change that solved a different bug of the "crash when no monitors are available" variety. I'm wondering if that fix affects this crash or if another fix is needed.

**Comment by haarp** [comment]
@murarth Is the change you're referring to EventLoop 2.0? I think I was already using it, due to using Alacritty 0.4.0. Sorry, I'm not very familiar with the interactions of glutin, winit and Alacritty.

Anyway, checking out winit master and running `cargo run --example window` gives me a window that will indeed not crash (but won't show anything either, giving a hall-of-mirrors effect)

**Comment by murarth** [comment]
Yes, winit example windows do look strange. They don't draw anything to the contents of the window, so they sometimes get filled with garbage data.

I'm referring to #1303, merged just 4 days ago. It prevents running window resize code (which leads to setting WM normal hints, which is the cause of this panic) when there are no monitors available.

If the winit example doesn't crash, I think it's safe to say that your issue has already been solved. If you want to test with Alacritty just to make sure, you can add a [`[patch]` section](https://doc.rust-lang.org/cargo/reference/manifest.html#the-patch-section) to the Cargo.toml of a working copy of alacritty. I think it would look something like this:

```toml
[patch.crates-io]
winit = { git = "https://github.com/rust-windowing/winit" }
```

Then, run `cargo update` and `cargo build`.
